tofu/gcp/modules/compute_instance/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "google_compute_instance" "this" {
+  boot_disk {
+    auto_delete = true
+
+    initialize_params {
+      image = var.boot_image
+      size  = 100
+      type  = "pd-standard"
+    }
+
+    mode = "READ_WRITE"
+  }
+
+  can_ip_forward      = true
+  deletion_protection = false
+  enable_display      = false
+
+  machine_type = var.machine_type
+  name         = var.instance_name
+
+  network_interface {
+    access_config {}
+
+    queue_count = 0
+    stack_type  = "IPV4_ONLY"
+    subnetwork  = var.network
```

Comment:
perhaps just an improvement for later;
this looks to be a pre-existing network name the user should specify. Is there a way this can create a new network if it doesn't exist?